File: message/message.go
package message

import "github.com/bpostlethwaite/cashpony/record"

type Smsg struct {
	Msg         string        `json:"msg"`
	LabelUpdate bool          `json:"labelupdate"`
	Flush       *chan Smsg    `json:"-"`
	Record      record.Record `json:"record"`
}

func (this *Smsg) HasRecInfo() bool {

	return !this.Record.IsEmpty()

}

File: web/static/js/message.js
function Message (spec) {

    if (!spec) spec = {};

    this.record = spec.record;
    this.labelupdate = (spec.labelupdate === undefined) ? false : spec.labelupdate;
    this.msg = spec.msg || '';
}

module.exports = Message;


Message.prototype.serialize = function () {

    function formatter (key, value) {
        if (key[0] === '_') return undefined;
        else return value;
    }

    return JSON.stringify(this, formatter, 2);
};
